### DISCUSSION TITLE
Future plans for custom themes

### DISCUSSION THREAD

**Discussion by soundsessential** [keycloak/keycloak#11598] [Archive]
Are there any plans for custom themes in the works? Primarily to move away from the Freemarker technology? I know there's lots of changes elsewhere and are these leading up to possible changes in the way custom themes are managed and created?

Many thanks

**Comment by xgp** [comment]
I also share the same question. Now that the account and admin v2 UI's both use React, the old, FTL way of overriding themes is less/not useful, and it's hard to override the new React themes without forking altogether. Would love to see some discussion around building component libraries (if React is indeed the future of Keycloak themes).

Also, I don't know if you are aware, but there is a great, community-contributed library for building login themes in React: https://github.com/InseeFrLab/keycloakify

**Reply by garronej** [reply]
Thank you for mentioning Keycloakify

**Comment by ssilvert** [comment]
@soundsessential We are definitely planning to make custom theming easier and it will almost certainly be done with React.  We will start doing R&D on this in Q3 after the new admin console becomes the default in Keycloak 19.

@xgp Thanks for the reference to keycloakify.  I hadn't seen that before.
